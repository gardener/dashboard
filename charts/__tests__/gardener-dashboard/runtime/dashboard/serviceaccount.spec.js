//
// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

'use strict'

const { helm } = fixtures

const renderTemplates = helm.renderDashboardRuntimeTemplates

describe('gardener-dashboard', function () {
  describe('serviceaccount', function () {
    let templates

    beforeEach(() => {
      templates = [
        'serviceaccount'
      ]
    })

    it('should not render the template with default values', async function () {
      const values = {}
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [serviceaccount] = documents
      expect(serviceaccount).toBe(null)
    })

    describe('when virtual garden is enabled', function () {
      it('should render the template', async function () {
        const values = {
          global: {
            virtualGarden: {
              enabled: true
            },
            dashboard: {
              serviceAccountTokenVolumeProjection: {
                enabled: true
              }
            }
          }
        }
        const documents = await renderTemplates(templates, values)
        expect(documents).toHaveLength(1)
        const [serviceaccount] = documents
        expect(serviceaccount).toMatchSnapshot()
      })

      it('should not render the template', async function () {
        const values = {
          global: {
            virtualGarden: {
              enabled: true
            },
            dashboard: {
              serviceAccountTokenVolumeProjection: {
                enabled: false
              }
            }
          }
        }
        const documents = await renderTemplates(templates, values)
        expect(documents).toHaveLength(1)
        const [serviceaccount] = documents
        expect(serviceaccount).toBe(null)
      })
    })
  })
})
