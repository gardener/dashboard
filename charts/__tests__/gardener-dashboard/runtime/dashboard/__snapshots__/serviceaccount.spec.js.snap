// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gardener-dashboard serviceaccount when virtual garden is enabled should render the template 1`] = `
Object {
  "apiVersion": "v1",
  "kind": "ServiceAccount",
  "metadata": Object {
    "labels": Object {
      "app.kubernetes.io/component": "dashboard",
      "app.kubernetes.io/instance": "gardener-dashboard",
      "app.kubernetes.io/managed-by": "Helm",
      "app.kubernetes.io/name": "gardener-dashboard",
      "helm.sh/chart": "gardener-dashboard-runtime-0.1.0",
    },
    "name": "gardener-dashboard",
    "namespace": "garden",
  },
}
`;
